### ISSUE TITLE
Field Prompt is required

### ISSUE THREAD

**Issue by latsami** [soonick/poMMo#124]
Hello,
I have used pommo many times using a form for people to subscribe from a website. I usually use a form generated by the admin interface.
But for some reason, when I try to subscribe, I have the error "Field Prompt" is required, a field I do not have. You can try it out here: http://esadgv3.stdin.fr/fr/contact/ 
Any idea?
Thanks!

**Comment by NelioRomao** [comment]
Hi,

input type="submit" value="S’abonner" name="pommo_signup"

i have been trying you Form and is very wired,, for what i can see you have a "  S'  " in you value
try just to leave that like this: value="Sabonner" or value="Subscribe"

Hope this can help.

**Comment by latsami** [comment]
Hi, thanks for your answer.
The apostrophe/single quotation mark in "S'abonner" shouldn't be a problem as I use double quotation marks around.
I did try to remove it, didn't do much; I also tried to put 'name="pommo_signup"' in the submit input instead of the hidden input, doesn't help either...

**Comment by NelioRomao** [comment]
Try use this link to make all the test insted of your form.
http://list.esadgv.stdin.fr/subscribe.php

The problem is the same. 

Try to delete and create all fields again, than you go to this link to test. http://list.esadgv.stdin.fr/subscribe.php

good luck this time

**Comment by soonick** [comment]
Hey latsami,

It's very hard to debug the way it is right now because I don't understand the language. Can you try setting a test page in English so I can take a look?

Thanks,
Adrian

**Comment by latsami** [comment]
fastbit, I can try this when I have a quiet time: the database has thousands of addresses and I don't want to mess things up.

Soonick, I have translated the form: http://esadgv3.stdin.fr/en/contact/
In the source code, only "nom" is not English, it means "name".

Then the error page (at http://list.esadgv.stdin.fr/process.php ) displays:

Configuration de l’abonnement
    Field Prompt est un champ requis.
← Retour au formulaire d’abonnement 

Subscription configuration
Field Prompt is a required field.
← Back to the subscription form

Thanks for your help.

**Comment by NelioRomao** [comment]
Can you send us a copy off your pommo with the email database clean.

Without this is very hard to help you with this erro.

**Comment by latsami** [comment]
Hello,
here is my pommo: http://stdin.fr/misc/latsami-pommo.tar.gz
I just removed the config file which had pwd info.
Hope you can find what's wrong, I really have no clue...
Thank you!

**Comment by NelioRomao** [comment]
Hi, 
Just make the download and the error is from Mail Exchanger 
"Setup/Configure/General", change to (PHP Mail Function)

It solve the problem.

Change this also:

classes/Pommo_Validate.php
     line number: 218
               function subscriberData(&$in, $p = array())
     Change to 
             public static function subscriberData(&$in, $p = array())

classes/Pommo_Helper_Messages.php
   line number 30
              function sendMessage($p = array('to' => false, 'type' => false, 'code' => false)) {
      Change to
              public static function sendMessage($p = array('to' => false, 'type' => false, 'code' => false)) {

Best Regards

![Capturar](https://f.cloud.github.com/assets/3768329/264909/df04d076-8d9f-11e2-957f-6335a7f98f28.PNG)

**Comment by latsami** [comment]
Hello,
thanks for taking a deep look into this. Only had time today to test it out. I changed everything you listed, and restarted apache (we never know) and still the Field Prompt issue.

And then I finally got the time to talk with my colleague about this. He is the one who set up the fields, and this missing "Field Prompt" is just the default value of the name of a subscribers field that he forgot to change.

In the end, I also realized that the "name" attribute in the input fields are quite obscure.
For example, for my two fields "fullname" and "group", instead of addressing them by their name, I have to use "d[3]" and "d[4]". It's weird, I don't remember it used to work that way? But maybe I just have a bad memory. Anyways, maybe it's relevant to change this in the future?

Thanks to both of you for your help. I'm a bit ashamed that I didn't see what was the real problem before....
